deployer/deployment_deployer.go

```diff
@@ -74,20 +74,15 @@ func (d *DeploymentDeployer) Deploy(ctx context.Context, dl *workloads.Deploymen
 		return errors.Wrap(err, "couldn't generate deployments data")
 	}
 
-	oldDeployments := d.tfPluginClient.StateLoader.currentNodeDeployment
-
-	currentDeployments, err := d.deployer.Deploy(ctx, oldDeployments, newDeployments, newDeploymentsSolutionProvider)
+	dl.NodeDeploymentID, err = d.deployer.Deploy(ctx, dl.NodeDeploymentID, newDeployments, newDeploymentsSolutionProvider)
 
 	// update deployment and plugin state
 	// error is not returned immediately before updating state because of untracked failed deployments
-	if currentDeployments[dl.NodeID] != 0 {
-		if dl.NodeDeploymentID == nil {
-			dl.NodeDeploymentID = make(map[uint32]uint64)
+	if dl.NodeDeploymentID[dl.NodeID] != 0 {
```

Comment:
while you are depending on the zero value of the map here, it broke the readability because the intention is not found not being zero.